src/test/java/com/thealgorithms/strings/AhoCorasickTest.java

```diff
@@ -0,0 +1,164 @@
+        final var expected = Map.of("GC", new ArrayList<Integer>(Arrays.asList(0)), "GCA", new ArrayList<Integer>(Arrays.asList(0)), "GCAT", new ArrayList<Integer>(Arrays.asList(0)));
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Check that the first occurrence of each pattern is at the beginning (position 0)
+        assertEquals(expected, positions);
+    }
+
+    /**
+     * Test searching for patterns that end at the end of the input text.
+     * The expected positions are 4, 3, and 2 for the patterns.
+     */
+    @Test
+    void testPatternAtEnd() {
+        // Define patterns that end at the end of the text
+        final var searchPatterns = new String[] {"CG", "TCG", "ATCG"};
+
+        // Define the expected positions (4, 3, 2) for the patterns
+        final var expected = new int[] {4, 3, 2};
+
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Check that the first occurrence of each pattern is at the expected position
+        for (int i = 0; i < searchPatterns.length; i++) {
+            assertEquals(expected[i], positions.get(searchPatterns[i]).get(0));
+        }
+    }
+
+    /**
+     * Test searching for patterns with multiple occurrences in the input text.
+     * The expected sizes are 1 and 1, and the expected positions are 2 and 3
+     * for the patterns "AT" and "T" respectively.
+     */
+    @Test
+    void testMultipleOccurrencesOfPattern() {
+        // Define patterns with multiple occurrences in the text
+        final var searchPatterns = new String[] {"AT", "T"};
+
+        // Define expected with two ArrayList of sizes (1, 1) and positions (2, 3) for the patterns
+        final var expected = Map.of("AT", new ArrayList<Integer>(Arrays.asList(2)), "T", new ArrayList<Integer>(Arrays.asList(3)));
+
+        // Perform the search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Check that the size and positions of each pattern match the expected values
+        assertEquals(expected, positions);
+    }
+
+    /**
+     * Test searching for patterns in a case-insensitive manner.
+     * The search should consider patterns regardless of their case.
+     */
+    @Test
+    void testCaseInsensitiveSearch() {
+        // Define patterns with different cases
+        final var searchPatterns = new String[] {"gca", "aTc", "C"};
+
+        // Perform the case-insensitive search using the Aho-Corasick algorithm
+        final var positions = AhoCorasick.search(text, searchPatterns);
+
+        // Check that the positions contain all the provided patterns in a case-insensitive way
+        for (String pattern : searchPatterns) {
+            assertTrue(positions.containsKey(pattern));
+        }
```

Comment:
Why not to define `expected` and check it it the same as `positions`?

Reply:
got it, implemented it.